### PR TITLE
QNetworks

### PR DESCRIPTION
**Pull request by Haichao-Zhang**
Implement QNetworks, which will be used by SAC (#398) etc.

### CODE REVIEW

File: alf/networks/q_networks.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        Args:
+            observation (torch.Tensor): consistent with `input_tensor_spec`
+            state: empty for API consistent with ValueRNNNetwork
+
+        Returns:
+            q_value (torch.Tensor): a tensor of the size (num_actions,)
+            state: empty
+        """
+        value = self._encoding_net(observation)
+        return torch.squeeze(value, -1), state
```
Comment (emailweixu) [inline alf/networks/q_networks.py]: why squeeze?

File: alf/networks/q_networks.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            observation (torch.Tensor): consistent with `input_tensor_spec`
+            state (nest[tuple]): a nest structure of state tuples (h, c)
+
+        Returns:
+            value (torch.Tensor): a 1D tensor
+            new_state (nest[tuple]): the updated states
+        """
+        encoding = self._encoding_net(observation)
+        q_value, state = self._lstm_encoding_net(encoding, state)
+        return torch.squeeze(q_value, -1), state
```
Comment (emailweixu) [inline alf/networks/q_networks.py]: and why squeeze?

File: alf/networks/q_networks.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+import torch.nn as nn
+
+from alf.networks import EncodingNetwork, LSTMEncodingNetwork
+import alf.layers as layers
+from alf.tensor_specs import TensorSpec, BoundedTensorSpec
+
+
+@gin.configurable
+class QNetwork(nn.Module):
+    """Create a q-network."""
```
[the remark this answers is not in the document]
Comment (Haichao-Zhang) [reply alf/networks/q_networks.py]: Comments are added for both q-networks and critic-networks.

File: alf/networks/q_networks.py (hunk 1 above)
Comment (Haichao-Zhang) [reply alf/networks/q_networks.py]: It's unnecessary. Removed.

File: alf/networks/q_networks.py (hunk 2 above)
Comment (Haichao-Zhang) [reply alf/networks/q_networks.py]: removed

**Pull request merged**